collections/src/main/scala/scalan/collections/Collections.scala

```diff
@@ -119,7 +120,7 @@ trait Collections { self: CollectionsDsl =>
 
     def singleton[T: Elem](v: Rep[T]): Coll[T] = {
       element[T] match {
-        case paE: CollectionElem[_, _] => ???
+        case paE: CollectionElem[_, _] => ???(s"Element is CollectionElem: $paE")
```

Comment:
`delayInvoke` method should be called. In this case invocation of `singleton` will be reified as `MethodCall`